.github/workflows/deploy-gh-pages.yaml

```diff
@@ -0,0 +1,41 @@
+name: Deploy to GitHub Pages
```

Comment:
@AA-Turner I think we don't deploy on the GitHub pages... What's the reason for this workflow?

Reply:
We don't currently. #1385 has extensive discussion on this (https://github.com/python/peps/pull/1385#issuecomment-687863467 etc) - the main drive is that deploying to GH pages is easy and free, and enables previewing the changes before any switchovers etc.

Reply:
Also discussion in various other issues (#2, #3, #25) and others I've likely forgoton (may be some on the pythondotorg repo)